server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2536,38 +2536,39 @@ private void doCheckIndex() throws IOException {
             return;
         }
         BytesStreamOutput os = new BytesStreamOutput();
-        PrintStream out = new PrintStream(os, false, StandardCharsets.UTF_8.name());
+        try (PrintStream out = new PrintStream(os, false, StandardCharsets.UTF_8.name())) {
```

Comment:
can we include the line above in try-with-resources block as well?

Reply:
I can add the instance of "BytesStreamOutput" in to the try-with-resources block, but it's just not a recommendation from CodeGuru. 😉

Reply:
After adding the instance of "BytesStreamOutput" into try-with-resources block, I found this error when running `gradle check`, so I reverted for now.
```
REPRODUCE WITH: ./gradlew ':plugins:repository-gcs:gcsThirdPartyTest' --tests "org.opensearch.repositories.gcs.GoogleCloudStorageThirdPartyTests.testCleanup" -Dtests.seed=DD99E92688B709B4 -Dtests.security.manager=false -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=et-EE -Dtests.timezone=Pacific/Pitcairn -Druntime.java=15

org.opensearch.repositories.gcs.GoogleCloudStorageThirdPartyTests > testCleanup FAILED
    RepositoryVerificationException[[test-repo] path [integration_test_third_party_tests_DD99E92688B709B4] is not accessible on master node]; nested: StorageException[Error getting access token for service account: Unexpected end of file from server]; nested: IOException[Error getting access token for service account: Unexpected end of file from server]; nested: SocketException[Unexpected end of file from server];
```

Reply:
Ok.. looks like the above error is not caused by that change. After I rebooted the machine, gradle check passed.

Reply:
I have included the line of `BytesStreamOutput os = new BytesStreamOutput();` into the try-with-resources block. 👍